### DISCUSSION TITLE
What Features in an ACE?

### DISCUSSION THREAD

**Discussion by DataBassGit** [daveshap/ACE_Framework#46] [General]
Looking for public opinions here. Let's get down to nuts and bolts. I want to talk about the individual functions/tools/features you think an ACE requires, should have, or would like to see. Please answer the following questions, but feel free to add commentary.

- What individual function or feature would you like to see in an individual ACE? 
- What lower level methods and logic make up those functions and features?
- Where do you think these pieces fit on the model?
- Do you think this is required, something that should be standard, or nice to have?

**Comment by daveshap** [comment]
Spontaneously learning and using arbitrary APIs e.g. gorilla, I think will be most useful. Kind of like the digital equivalent of human hands and eyes (human hands and eyes can interact with any number of external devices and information). Arbitrary API adoption will make ACEs robust and general purpose. Consider that those APIs could be internet, LAN, or even for hardware and sensors.
